generic/release-checklist.md

```diff
@@ -0,0 +1,103 @@
+---
+# Template generated by https://github.com/coreos/repo-templates; do not edit downstream
+---
+
+# Release process
+
+The release process follows the usual PR-and-review flow, allowing an external reviewer to have a final check before publishing.
+
+## Requirements
+
+This guide requires:
+
+ * A web browser (and network connectivity)
+ * `git`
+ * [GPG setup][GPG setup] and personal key for signing
+ * Write access to this GitHub project
+
+## Release checklist
+
+These steps show how to release version `x.y.z` on the `origin` remote (this can be checked via `git remote -av`).
+Push access to the upstream repository is required in order to publish the new tag and the PR branch.
+
+:warning:: if `origin` is not the name of the locally configured remote that points to the upstream git repository (i.e. `git@github.com:coreos/{{ git_repo }}.git`), be sure to assign the correct remote name to the `UPSTREAM_REMOTE` variable.
+
+{% if do_release_notes_doc %}
+- write release notes
+  - [ ] write release notes in `docs/release-notes.md`; get them reviewed and merged
+  - [ ] if doing a branched release, also include a PR to merge the `docs/release-notes.md` changes into main
+{% endif %}
+
+- make sure the project is clean and prepare the environment:
+  - [ ] `git clean -fd`
+  - [ ] `RELEASE_VER=x.y.z`
+  - [ ] `UPSTREAM_REMOTE=origin`
+
+- create release commit on a dedicated branch and tag it (the commit and tag will be signed with the GPG signing key you configured):
+  - [ ] `git checkout -b release-${RELEASE_VER}`
+
+- open and merge a PR for this release:
+  - [ ] `git push ${UPSTREAM_REMOTE} release-${RELEASE_VER}`
+  - [ ] open a web browser and create a PR for the branch above
+  - [ ] make sure the resulting PR contains exactly one commit
+{%- if not do_release_notes_doc %}
+  - [ ] in the PR body, write a short changelog with relevant changes since last release
+{%- endif %}
+  - [ ] get the PR reviewed, approved and merged
+
+- publish the artifacts (tag):
+  - [ ] `git checkout v${RELEASE_VER}`
+  - [ ] `git push ${UPSTREAM_REMOTE} v${RELEASE_VER}`
```

Comment:
This was copied from a release flow that automatically generated the tag.  The tag is not currently being created anywhere in this procedure.  We should also add some cross-checking to make sure the tag was created on the correct commit, since errors cannot be fixed without abandoning the release and starting over.